### PR TITLE
2021.4 updates Dockerfile

### PR DESCRIPTION
**Pull request by ryanloney**
Locked the openvino-dev==2021.4 so we can tag a 2021.4 release with this Dockerfile. It is also updated to avoid double copying the notebooks repo into the container.

### CODE REVIEW

**Review by helena-intel — CHANGES_REQUESTED** [review]
Thanks Ryan! Very nice that we can now include openvino-dev in the requirements.in without conflicts. If you want to move forward, we can ignore the tensorflow comments for now, but I would like to address the "Building a docker image locally doesn't include the notebooks anymore" issue before merging.

File: Dockerfile [hunk 1]
```diff
@@ -37,9 +37,7 @@ RUN chown -R 1001:0 /tmp/scripts /tmp/src
 USER 1001
 RUN /tmp/scripts/assemble
 
-COPY notebooks .
-
-RUN pip install openvino-dev tensorflow-serving-api nbval grpcio
+RUN pip install tensorflow-serving-api
```
[the remark this answers is not in the document]
Comment (ryanloney) [reply Dockerfile:40]: I could not get it to work. There is an issue with 2.5 in pipenv and trying to install keras by dependency. I spent a few hours and kept hitting the same problem.

File: Dockerfile [hunk 2]
```diff
@@ -37,9 +37,7 @@ RUN chown -R 1001:0 /tmp/scripts /tmp/src
 USER 1001
 RUN /tmp/scripts/assemble
 
-COPY notebooks .
```
[the remark this answers is not in the document]
Comment (helena-intel) [reply Dockerfile:40]: Previous comment was misunderstanding: the notebooks are not copied to the image on build anymore, but on `docker run` git clone clones the repo and the notebooks will appear.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (helena-intel) [reply Dockerfile:40]: Yikes. Let's leave it like this and please do not spend hours in dependency hell again!

**Review by helena-intel — APPROVED** [review]

**Pull request merged**